### PR TITLE
Resolve issue#1586 | Update GameLift server packaging page to address out-of-date install script and add release build packaging

### PR DESCRIPTION
**Pull request by chanmosq**
### Submission Checklist:

* [n/a] **Descriptive active voice** - Do descriptive sentences have a clear *subject* and *action verb*?
* [n/a] **Answer the question at hand** - Does the documentation answer a *what*, *why*, *how*, or *where* type of question?
* [n/a] **Consistency** - Does the content consistently follow the [style guide](https://o3de.org/docs/contributing/to-docs/style-guide/)?
* [x] **Help the user** - Does the documentation show the user something *meaningful*?

### CODE REVIEW

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/advanced-topics.md [hunk 1]
```diff
@@ -134,6 +134,6 @@ C:\game\bin\server.exe --engine-path=C:\game --project-path=C:\game\process1 --p
 
 ---
 
-Previous topic: [AWS GameLift Gem C++ API](cpp-api/)
+Previous topic: [FlexMatch Scripting](flexmatch/scripting/)
```
Comment (willihay) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/advanced-topics.md:137]: I'm not sure why this section has manually defined "Previous topic" and "Next topic" links, but I think we should remove all these in favor of some sort of automation in the page design. Might be worth a GHI or two if you feel like making them (e.g. one to remove the manual links and another to design built-in "previous/next" links on the page).

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/build-packaging-for-linux.md [hunk 2]
```diff
@@ -58,40 +59,42 @@ You must create a separate installation folder to copy over the required assets,
...
-  
+
+    sudo cp -a ./lib64/libX* /lib64/.
+    sudo cp -a ./lib64/libbsd* /lib64/.
+
     if cat /etc/system-release | grep -qFe 'Amazon Linux release 2'
     then
-     sudo yum groupinstall 'Development Tools' -y
-     sudo yum install python3 -y
+    sudo yum update -y
```
Comment (willihay) [inline content/docs/user-guide/gems/reference/aws/aws-gamelift/build-packaging-for-linux.md:68]: Did you intentionally remove the line indentation inside the `then` clause? Might want to put that back for readability. (And maybe use two spaces - I think it was only one space before.)

**Review by willihay — APPROVED** [review]
Approved with suggestions.

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/advanced-topics.md (hunk 1 above)
Comment (chanmosq) [reply content/docs/user-guide/gems/reference/aws/aws-gamelift/advanced-topics.md:137]: Yes, I was debating what to do about next/previous links as well. For this section (GameLift), I recall incoming PRs specifically defining these links, and as they were consistent across other pages in this section, I decided to keep them in. 

I do lean towards voting to remove the links, as it's not something we have in other sections. Nor do I think it helps with navigation for this section, as the pages can be viewed in different orders. And as shown in this PR, they can easily become outdated. I can create an issue for that. 

As for adding built-in previous/next navigation links, I can create an issue for that to begin and track a discussion regarding whether or not we want that feature.

**Review by junbo75 — APPROVED** [review]
Looks good to me. Thanks for the update!

File: content/docs/user-guide/gems/reference/aws/aws-gamelift/build-packaging-for-linux.md (hunk 2 above)
Comment (chanmosq) [reply content/docs/user-guide/gems/reference/aws/aws-gamelift/build-packaging-for-linux.md:68]: Thanks for catching, I fixed the indentation!

**Pull request merged**